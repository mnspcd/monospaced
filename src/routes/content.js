module.exports = {
  title: "Monospaced",
  description: "Modern web software solutions",
  blog: [], // require("../blog").default,
  clients: [
    {
      href: "http://bladudflies.com",
      icon: "BladudFlies",
    },
    {
      href: "https://www.compass-group.co.uk",
      icon: "Chartwells",
    },
    {
      href: "https://www.digitas.com",
      icon: "Digitas",
    },
    {
      href: "https://www.dowjones.com",
      icon: "DowJones",
    },
    {
      href: "https://www.findmypast.co.uk",
      icon: "FindMyPast",
    },
    {
      href: "https://frfrm.com",
      icon: "Freeform",
    },
    {
      href: "http://www.thegrandunion.com",
      icon: "GrandUnion",
    },
    {
      href: "https://hanno.co",
      icon: "Hanno",
    },
    {
      href: "https://www.majorplayers.co.uk",
      icon: "MajorPlayers",
    },
    {
      href: "https://www.national-lottery.co.uk",
      icon: "NationalLottery",
    },
    {
      href: "https://www.nhs.uk",
      icon: "Nhs",
    },
    {
      href: "https://www.ogilvy.com",
      icon: "Ogilvy",
    },
    {
      href: "https://www.social360monitoring.com",
      icon: "Social360",
    },
    {
      href: "https://www.stordis.com",
      icon: "Stordis",
    },
    {
      href: "http://wearesuburb.com",
      icon: "Suburb",
    },
    {
      href: "https://thomsonreuters.com",
      icon: "ThomsonReuters",
    },
  ],
  footerLinks: [
    {
      href: "/legal/",
      routerLink: true,
      text: "legal",
    },
    {
      href: "/lighthouse.report.html",
      text: "lighthouse",
    },
    {
      href: "/login/",
      text: "login",
    },
    {
      href: "http://localhost/storybook/",
      text: "storybook",
    },
    {
      href: "https://github.com/monospaced/monospaced.com",
      text: "view-source",
    },
  ],
  mastheadLinks: [
    /* {
      href: "/blog/",
      routerLink: true,
      text: "blog",
    },
    {
      href: "/design-system/",
      routerLink: true,
      text: "design-system",
    }, */
    {
      href: "mailto:info@monospaced.com",
      text: "info@monospaced.com",
    },
  ],
};
