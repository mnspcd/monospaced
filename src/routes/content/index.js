import {
  BladudFlies,
  Chartwells,
  Digitas,
  DowJones,
  FindMyPast,
  Freeform,
  GrandUnion,
  Hanno,
  MajorPlayers,
  NationalLottery,
  Nhs,
  Ogilvy,
  Social360,
  Stordis,
  Suburb,
  ThomsonReuters,
} from "./svg";

export default {
  blog: {
    heading: "Blog",
    posts: [], // require("./blog").default,
  },
  clients: {
    heading: "Selected clients",
    list: [
      {
        href: "http://bladudflies.com",
        label: "Bladud Flies!",
        Logo: BladudFlies,
      },
      {
        href: "https://www.compass-group.co.uk",
        label: "Chartwells",
        Logo: Chartwells,
      },
      {
        href: "https://www.digitas.com",
        label: "Digitas",
        Logo: Digitas,
      },
      {
        href: "https://www.dowjones.com",
        label: "Dow Jones",
        Logo: DowJones,
      },
      {
        href: "https://www.findmypast.co.uk",
        label: "Find My Past",
        Logo: FindMyPast,
      },
      {
        href: "https://frfrm.com",
        label: "Freeform Labs",
        Logo: Freeform,
      },
      {
        href: "http://www.thegrandunion.com",
        label: "Grand Union",
        Logo: GrandUnion,
      },
      {
        href: "https://hanno.co",
        label: "Hanno",
        Logo: Hanno,
      },
      {
        href: "https://www.majorplayers.co.uk",
        label: "Major Players",
        Logo: MajorPlayers,
      },
      {
        href: "https://www.national-lottery.co.uk",
        label: "The National Lottery",
        Logo: NationalLottery,
      },
      {
        href: "https://www.nhs.uk",
        label: "NHS",
        Logo: Nhs,
      },
      {
        href: "https://www.ogilvy.com",
        label: "Ogilvy",
        Logo: Ogilvy,
      },
      {
        href: "http://www.social360monitoring.com",
        label: "Social 360",
        Logo: Social360,
      },
      {
        href: "https://www.stordis.com",
        label: "Stordis",
        Logo: Stordis,
      },
      {
        href: "http://wearesuburb.com",
        label: "Suburb",
        Logo: Suburb,
      },
      {
        href: "https://www.thomsonreuters.com",
        label: "Thomson Reuters",
        Logo: ThomsonReuters,
      },
    ],
  },
  footer: {
    links: [
      {
        href: "/legal/",
        routerLink: true,
        text: "legal",
      },
      // {
      //   href: "/lighthouse.report.html",
      //   text: "lighthouse",
      // },
      {
        href: "/login/",
        text: "login",
      },
      {
        href: "https://github.com/mnspcd/monospaced.com",
        text: "view-source",
      },
    ],
  },
  legal: {
    company: {
      content: require("./legal/company.mdx").default,
      heading: "Company details",
    },
    heading: "Legal matters",
    privacy: {
      content: require("./legal/privacy.mdx").default,
      heading: "Privacy policy",
    },
  },
  masthead: {
    links: [
      /* {
        href: "/blog/",
        routerLink: true,
        text: "blog",
      }, */
      {
        href: "/design-system/?path=/docs/introduction--page",
        text: "design-system",
      },
      {
        href: "mailto:hello@monospaced.com",
        text: "hello@monospaced.com",
      },
    ],
  },
  meta: {
    description: "Modern web software solutions",
    title: "Monospaced",
  },
  services: {
    heading: "Our services",
    list: [
      "Design systems engineering",
      "Front-end development",
      "Legacy code rescue",
      "Prototyping, concepts & MVPs",
      "Web applications",
    ],
  },
};
