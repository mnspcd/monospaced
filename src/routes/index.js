import { MDXProvider } from "@mdx-js/react";
import PropTypes from "prop-types";
import React from "react";
import { IndexRoute, Route } from "react-router";

import Footer from "../components/Footer";
import { components } from "../components/Markdown";
import content from "./content";

import Blog, { Post } from "./Blog";
import Home from "./Home";
import Legal from "./Legal";
import NotFound from "./NotFound";

const App = ({
  children,
  route: {
    content: { title, footerLinks },
  },
  routes,
}) => {
  return (
    <MDXProvider components={components}>
      <>
        {React.cloneElement(children, { content }, routes)}
        <Footer copyright={title} links={footerLinks} routes={routes} />
      </>
    </MDXProvider>
  );
};

const forceTrailingSlash = (nextState, replace) => {
  const path = nextState.location.pathname;
  if (path.slice(-1) !== "/") {
    replace({ ...nextState.location, pathname: `${path}/` });
  }
};

const forceTrailingSlashOnChange = (prevState, nextState, replace) => {
  forceTrailingSlash(nextState, replace);
};

const Routes = (
  <Route onChange={forceTrailingSlashOnChange} onEnter={forceTrailingSlash}>
    <Route component={App} content={content} path="/">
      <IndexRoute component={Home} />
      {content.blog.length && <Route component={Blog} path="blog" />}
      {content.blog.map(post => {
        const { slug } = post;
        return <Route component={Post} key={slug} path={slug} />;
      })}
      <Route component={Legal} path="legal" />
      <Route component={NotFound} path="404" />
      <Route component={NotFound} path="*" />
    </Route>
  </Route>
);

App.propTypes = {
  children: PropTypes.node,
  route: PropTypes.object,
  routes: PropTypes.array,
};

export default Routes;
