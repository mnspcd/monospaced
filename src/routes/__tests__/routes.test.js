/* eslint-env jest */
import { createMemoryHistory, Router } from "react-router";
import React from "react";
import reactRouterToArray from "react-router-to-array";
import renderer from "react-test-renderer";

import routes from "../../routes";

jest.mock("../content");

describe("Routes", () => {
  it("should convert correctly to an array", () => {
    const array = renderer.create(reactRouterToArray(routes)).toJSON();
    expect(array).toMatchSnapshot();
  });

  it("should force trailing slashes on urls", () => {
    const history = createMemoryHistory("/foo");
    renderer.create(<Router history={history} routes={routes} />);
    expect(history.getCurrentLocation().pathname).toBe("/foo/");
    history.push("/bar");
    expect(history.getCurrentLocation().pathname).toBe("/bar/");
  });
});
