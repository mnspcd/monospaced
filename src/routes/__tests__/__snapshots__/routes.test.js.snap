// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes should render App correctly 1`] = `
<div>
  <div
    class="mdrn"
  >
    <div
      content="[object Object]"
    />
    <div
      class="Footer"
    >
      <div
        class="Footer-copyright"
      >
        Â© 
        foo
         
        2018
      </div>
      <ul
        class="Footer-list"
      />
    </div>
  </div>
</div>
`;
